extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -142,6 +144,24 @@ export class AmpAdXOriginIframeHandler {
       });
     }
 
+    // Wait for initial load signal. Notice that this signal is not
+    // used to resolve the final layout promise because iframe may still be
+    // consuming significant network and CPU resources.
+    listenForOncePromise(this.iframe, CommonSignals.INI_LOAD, true).then(() => {
+      // TODO(dvoytenko, #7788): ensure that in-a-box "ini-load" message is
+      // received here as well.
+      this.baseInstance_.signals().signal(CommonSignals.INI_LOAD);
+    });
+
+    if (opt_isA4A) {
+      // A4A writes creative frame directly to page once creative is received
+      // and therefore does not require render start message so attach and
+      // impose no loader delay.  Network is using renderStart or
+      // bootstrap-loaded to indicate ad request was sent, either way we know
+      // that occurred for Fast Fetch.
+      this.element_.appendChild(this.iframe);
+      return Promise.resolve();
```

Comment:
I commented on the thread. This is not ideal. `layoutCallback` should return the promise most representative of loading time. To update rendering state we should simply call `renderStart_`. E.g.

```
this.element_.appendChild(this.iframe);   // THIS IS AN EXISTING LINE
if (opt_isA4A) {
  this.renderStart_(info);
  renderStartResolve();
}
```

The same can be arranged via micro-tasks if it's more desirable to keep all `render-start` code together up top.